data/datasets/credit_card_default_loader.py

```diff
@@ -0,0 +1,88 @@
+from typing import Optional
+import pandas as pd
+from data.datasets import base_loader
+from data.datasets import utils
+
+
+DATASET_NAME = "credit_card_default"
+_SOURCE_TARGET_COLUMN = "default payment next month"
+_URL = (
+    "https://archive.ics.uci.edu/ml/machine-learning-databases/00350/"
+    "default%20of%20credit%20card%20clients.xls"
+)
+
+
+class CreditCardDefaultLoader(base_loader.DataLoader):
+    """Loads the UCI Credit Card Default dataset from the internet or local
+    storage.
+
+    The dataset has columns:
+        LIMIT_BAL: The amount of the given credit.
+        SEX: Binary variable where 1 is Male and 2 is Female.
+        EDUCATION: A categorical variable. The dataset source includes
+            descriptions for values 1, 2, 3, and 4, but does not explain values
+            0, 5, and 6 which are also included in the data. The column is
+            dropped for this reason.
+        MARRIAGE: A categorical variable. The dataset source includes
+            descriptions for values 1, 2, and 3, but does not explain value 0
+            which is also incldued in the data. The column is dropped for this
+            reason.
+        AGE: Numerical age in years.
+        PAY_[1-6]: A values -2, -1, and 0 indicate that the bill was paid on
+            on time. Values 1 through 9 indicate how many months late the
+            payment was.
+        BILL_AMT[1-6]
+        PAY_AMT[1-6]"""
```

Comment:
Missing descriptions of BILL_AMT and PAY_AMT, for consistency. It could be useful to state in words/explicitly here that there are 6 columns for PAY, BILL_AMT, and PAY_AMT (indexed as PAY_i, BILL_AMT_i, etc.).